metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -88,25 +89,33 @@ final case class Indexer(
       buildTool: BuildTool,
       checksum: String,
       importBuild: BspSession => Future[Unit],
+      reconnectToBuildServer: () => Future[BuildChange],
   ): Future[BuildChange] = {
     def reloadAndIndex(session: BspSession): Future[BuildChange] = {
       workspaceReload().persistChecksumStatus(Status.Started, buildTool)
 
-      session
-        .workspaceReload()
-        .flatMap(_ => importBuild(session))
-        .map { _ =>
-          scribe.info("Correctly reloaded workspace")
-          profiledIndexWorkspace(doctorCheck)
-          workspaceReload().persistChecksumStatus(Status.Installed, buildTool)
-          BuildChange.Reloaded
-        }
-        .recoverWith { case NonFatal(e) =>
-          scribe.error(s"Unable to reload workspace: ${e.getMessage()}")
-          workspaceReload().persistChecksumStatus(Status.Failed, buildTool)
-          languageClient.showMessage(Messages.ReloadProjectFailed)
-          Future.successful(BuildChange.Failed)
-        }
+      buildTool match {
+        case _: BspOnly => reconnectToBuildServer()
```

Comment:
```suggestion
        case _: BspOnly if !session.main.canReload() => reconnectToBuildServer()
```
we would also need to add cnaReload method.

Reply:
I'm not sure that makes a lot of sense. If `BspOnly` changes, that means the bsp config changes (so args for starting the server). I can't really imagine a server would take those changes into account on reload.

Reply:
Is it not possible that it will also happen via didChangeBuildTarget notification? We do reload in that case I think

Reply:
Yes, we do. But it's not that method actually.

Reply:
Ach, you're right. Nvm me then.